### PR TITLE
docs: update godoc uses of "provider" to "driver" or "service" in top-level documentation

### PR DESCRIPTION
**Pull request by vangent**
Updates #2548.

This updates two top-level doc files; I pulled it out separately since they are higher visibility.

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -21,15 +21,15 @@ It does this by providing stable, idiomatic interfaces for common uses like
 storage and databases. Think `database/sql` for cloud products.
 
 At the core of the Go CDK are common "portable types" implemented by cloud
-providers. For example, objects of the blob.Bucket portable type can be created
-using gcsblob.OpenBucket, s3blob.OpenBucket, or any other provider. Then, the
-blob.Bucket can be used throughout your application without worrying about
-the underlying implementation.
+services. For example, objects of the blob.Bucket portable type can be created
```
Comment (polinasok) [inline doc.go]: "portable types". These are service-independent abstractions for interacting with any supported cloud services that have a corresponding driver implementation.
OR
"portable types" implemented for supported cloud services by service-dependent drivers.

File: doc.go [hunk 2]
```diff
@@ -21,15 +21,15 @@ It does this by providing stable, idiomatic interfaces for common uses like
...
+using gcsblob.OpenBucket, s3blob.OpenBucket, or any other service with a
+Go CDK driver implementation. Then, the blob.Bucket can be used throughout
```
Comment (polinasok) [inline doc.go]: And with the change above, this can be simplified to just "or any other driver".

File: doc.go (hunk 1 above)
Comment (vangent) [reply doc.go]: Reworded a bit, PTAL.

File: doc.go (hunk 2 above)
Comment (vangent) [reply doc.go]: Done.

File: doc.go [hunk 3]
```diff
@@ -20,11 +20,12 @@ seamlessly deploy cloud applications on any combination of cloud providers.
 It does this by providing stable, idiomatic interfaces for common uses like
 storage and databases. Think `database/sql` for cloud products.
 
-At the core of the Go CDK are common "portable types" implemented by cloud
-services. For example, objects of the blob.Bucket portable type can be created
-using gcsblob.OpenBucket, s3blob.OpenBucket, or any other service with a
-Go CDK driver implementation. Then, the blob.Bucket can be used throughout
-your application without worrying about the underlying implementation.
+At the core of the Go CDK are common "portable types", implemented on top of
+service-specific drivers for supported supported cloud services. For example,
```
Comment (polinasok) [inline doc.go]: remove 2nd "supported"

Comment (vangent) [reply doc.go]: Oops, done.

**Pull request merged**